### ISSUE TITLE
Implement the use of `&` to background a command

### ISSUE THREAD

**Issue by jpsnover** [PowerShell/PowerShell#716]
So we should implement it.

**Comment by pirate** [comment]
Apart from seeming a bit hacky, are there real, practical downsides to implementing `&` in fish using bash as a subshell spawner?
```bash
function background --description "Run a fish command in the background using a bash subshell"
	bash -c "fish -c '""$argv""' &"
end

background "sleep 5; echo hi"
```
This `background` implementation appears to have exactly the behavior I'd expect from native `&`.

**Comment by jombooth** [comment]
@pirate yes; the process launched by "background" will not have access to unexported variables or user-defined functions in fish. Same would apply to a PowerShell implementation that worked this way.

```shell
user@user /> function background
                        bash -c "fish -c '""$argv""' &"
                      end

user@user /> function sayhello
                        echo hello
                      end

user@user /> sayhello
hello
user@user /> background sayhello
... fish: Unknown command “sayhello”
Standard input: sayhello
```

**Comment by BrucePay** [comment]
This has been merged #3360

**Comment by oising** [comment]
Looking forward to trying this out!

Sent from my Windows 10 phone
